MantidQt/SliceViewer/src/CoordinateTransform.cpp

```diff
@@ -23,34 +23,37 @@ void NullTransform::checkDimensionsForHKL(Mantid::API::IMDWorkspace_sptr ws,
 NonOrthogonalTransform::~NonOrthogonalTransform() {}
 
 NonOrthogonalTransform::NonOrthogonalTransform(
-    Mantid::API::IMDWorkspace_sptr ws, size_t dimX, size_t dimY)
-    : m_dimensionsHKL(true) {
+    const Mantid::API::IMDWorkspace_const_sptr &workspace, size_t dimX,
+    size_t dimY)
+    : m_dimensionsHKL(true), m_skewMatrix({0}) {
   // Set the skewMatrix for the non-orthogonal data
-  auto numberOfDimensions = ws->getNumDims();
+  auto numberOfDimensions = workspace->getNumDims();
```

Comment:
A reference would better describe ownership, but I also understand the desire to keep pull requests small. Perhaps it's best to make a note of it in a separate issue?